### DISCUSSION TITLE
upgrading from 4.0.3 to 4.1.0 netbox-rq failes to start

### DISCUSSION THREAD

**Discussion by TalkBackJohnny** [netbox-community/netbox#17403] [Help Wanted!] [answered]
upgrading from version 4.0.3 to 4.1.0 using the git process - there don't seem to be any errors running the upgrade.sh script we have a basic install and only adding LDAP config - Ubuntu 22.04

systemctl status netbox-rq
● netbox-rq.service - NetBox Request Queue Worker
     Loaded: loaded (/etc/systemd/system/netbox-rq.service; enabled; vendor preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Fri 2024-09-06 18:19:37 UTC; 6s ago
       Docs: https://docs.netbox.dev/
    Process: 13731 ExecStart=/opt/netbox/bin/python3 /opt/netbox/netbox/manage.py rqworker high default low (code=exited, status=1/FAILURE)
   Main PID: 13731 (code=exited, status=1/FAILURE)
        CPU: 10.280s

 journalctl -eu netbox-rq

Sep 06 18:50:02 wb-netbox-t02 python3[16712]: Traceback (most recent call last):
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/opt/netbox/netbox/manage.py", line 10, in <module>
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     execute_from_command_line(sys.argv)
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/usr/local/lib/python3.10/dist-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     utility.execute()
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/usr/local/lib/python3.10/dist-packages/django/core/management/__init__.py", line 436, in execute
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     self.fetch_command(subcommand).run_from_argv(self.argv)
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/usr/local/lib/python3.10/dist-packages/django/core/management/base.py", line 413, in run_from_argv
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     self.execute(*args, **cmd_options)
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/usr/local/lib/python3.10/dist-packages/django/core/management/base.py", line 454, in execute
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     self.check()
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/usr/local/lib/python3.10/dist-packages/django/core/management/base.py", line 486, in check
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     all_issues = checks.run_checks(
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/usr/local/lib/python3.10/dist-packages/django/core/checks/registry.py", line 88, in run_checks
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     new_errors = check(app_configs=app_configs, databases=databases)
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/usr/local/lib/python3.10/dist-packages/django/core/checks/urls.py", line 42, in check_url_namespaces_unique
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     all_namespaces = _load_all_namespaces(resolver)
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/usr/local/lib/python3.10/dist-packages/django/core/checks/urls.py", line 61, in _load_all_namespaces
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     url_patterns = getattr(resolver, "url_patterns", [])
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/usr/local/lib/python3.10/dist-packages/django/utils/functional.py", line 47, in __get__
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     res = instance.__dict__[self.name] = self.func(instance)
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/usr/local/lib/python3.10/dist-packages/django/urls/resolvers.py", line 738, in url_patterns
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/usr/local/lib/python3.10/dist-packages/django/utils/functional.py", line 47, in __get__
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     res = instance.__dict__[self.name] = self.func(instance)
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/usr/local/lib/python3.10/dist-packages/django/urls/resolvers.py", line 731, in urlconf_module
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     return import_module(self.urlconf_name)
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     return _bootstrap._gcd_import(name[level:], package, level)
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/opt/netbox/netbox/netbox/urls.py", line 10, in <module>
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     from netbox.graphql.schema import schema
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/opt/netbox/netbox/netbox/graphql/schema.py", line 37, in <module>
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     schema = strawberry.Schema(
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/usr/local/lib/python3.10/dist-packages/strawberry/schema/schema.py", line 145, in __init__
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     self._schema = GraphQLSchema(
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/usr/local/lib/python3.10/dist-packages/graphql/type/schema.py", line 224, in __init__
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     collect_referenced_types(query)
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/usr/local/lib/python3.10/dist-packages/graphql/type/schema.py", line 433, in collect_referenced_types
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     collect_referenced_types(field.type)
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/usr/local/lib/python3.10/dist-packages/graphql/type/schema.py", line 432, in collect_referenced_types
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     for field in named_type.fields.values():
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/usr/lib/python3.10/functools.py", line 981, in __get__
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     val = self.func(instance)
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:   File "/usr/local/lib/python3.10/dist-packages/graphql/type/definition.py", line 811, in fields
Sep 06 18:50:02 wb-netbox-t02 python3[16712]:     raise cls(f"{self.name} fields cannot be resolved. {error}") from error
Sep 06 18:50:02 wb-netbox-t02 python3[16712]: TypeError: EventRuleType fields cannot be resolved. GraphQL type for model field 'extras.EventRule.event_types' has not been implemented
Sep 06 18:50:03 wb-netbox-t02 systemd[1]: netbox-rq.service: Main process exited, code=exited, status=1/FAILURE
Sep 06 18:50:03 wb-netbox-t02 systemd[1]: netbox-rq.service: Failed with result 'exit-code'.
Sep 06 18:50:03 wb-netbox-t02 systemd[1]: netbox-rq.service: Consumed 10.185s CPU time.

**Comment by candlerb** [comment, marked as answer]
> Sep 06 18:50:02 wb-netbox-t02 python3[16712]: File "/usr/local/lib/python3.10/dist-packages/django/core/management/init.py", line 442, in execute_from_command_line

You're running with global Python libraries. Don't do that; Netbox must always run in its own virtualenv.

> Process: 13731 ExecStart=/opt/netbox/bin/python3 /opt/netbox/netbox/manage.py rqworker high default low (code=exited, status=1/FAILURE)

What is directory /opt/netbox/bin/ for? This shouldn't exist. You should be using /opt/netbox/venv/bin/python3 to run python inside the venv - as per the [supplied config file](https://github.com/netbox-community/netbox/blob/v4.0.8/contrib/netbox-rq.service).

The installation instructions are there to help you. If you decide to install things differently, and things break, then it's up to you to debug and fix any problems.

**Reply by TalkBackJohnny** [reply]
That actually helped to point me in the correct direction and fix the issues I had with my deploy. Cleaned things up and am now good to go.
